src/osdc/Journaler.cc

```diff
@@ -621,7 +621,7 @@ void Journaler::_do_flush(unsigned amount)
   assert(!readonly);
 
   // flush
-  uint64_t len = write_pos - flush_pos;
+  int64_t len = write_pos - flush_pos;
```

Comment:
A negative value here would immediately assert out on the following line because write_buf.length() cannot be negative -- it seems misleading to used a signed value here.

Also, we already asserted at the top of the function that write_pos must be greater than flush_pos.

Using a signed integer here actually seems *less* correct, because it would not work properly in the hypothetical scenario where write_pos was so great that it could not be represented in a signed form.